src/main/java/graphql/execution/AsyncExecutionStrategy.java

```diff
@@ -0,0 +1,92 @@
+package graphql.execution;
+
+import graphql.ExecutionResult;
+import graphql.ExecutionResultImpl;
+import graphql.language.Field;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.function.BiConsumer;
+
+/**
+ * The standard graphql execution strategy that runs fields asynchronously
+ */
+public class AsyncExecutionStrategy extends ExecutionStrategy {
```

Comment:
We need to explain to people that DataFetchers can returns CFs and that they will be chained ok.

That needs to go on DataFetcher and in the general documentation.

I think the doco should have a "aasyncrhnous support in graphql" section (but not in this PR)